shopfloor_reception/services/reception.py

```diff
@@ -0,0 +1,1155 @@
+            dest_location = selected_line.location_dest_id
+            child_locations = self.env["stock.location"].search(
+                [("id", "child_of", dest_location.id)]
+            )
+            pack_location = package.location_id
+            if pack_location:
+                if pack_location not in child_locations:
+                    # If the scanned package has a location that isn't a child
+                    # of the move dest, return an error
+                    message = self.msg_store.dest_location_not_allowed()
+                    return self._response_for_set_quantity(
+                        picking, selected_line, message=message
+                    )
+                else:
+                    # If the scanned package has a valid destination,
+                    # set both package and destination on the package,
+                    # and go back to the selection line screen
+                    selected_line.result_package_id = package
+                    selected_line.location_dest_id = pack_location
+                    return self._response_for_select_line(picking)
+            # Scanned package has no location, move to the location selection
+            # screen
+            selected_line.result_package_id = package
+            return self._response_for_set_destination(picking, selected_line)
+
+    def _set_quantity__by_location(self, picking, selected_line, barcode):
+        search = self._actions_for("search")
+        location = search.location_from_scan(barcode)
+        if location:
+            dest_location = selected_line.location_dest_id
+            child_locations = self.env["stock.location"].search(
+                [("id", "child_of", dest_location.id)]
+            )
+            if location not in child_locations:
+                # Scanned location isn't a child of the move's dest location
+                message = self.msg_store.dest_location_not_allowed()
+                return self._response_for_set_quantity(
+                    picking, selected_line, message=message
+                )
+            # process without pack, set destination location, and go back to
+            # `select_line`
+            selected_line.location_dest_id = location
+            return self._response_for_select_line(picking)
+
+    def _use_handlers(self, handlers, *args, **kwargs):
+        for handler in handlers:
+            response = handler(*args, **kwargs)
+            if response:
+                return response
+
+    def _assign_user_to_picking(self, picking):
+        picking.user_id = self.env.user
+
+    # DATA METHODS
+
+    def _data_for_stock_picking(self, picking, with_lines=False):
+        data = self.data.picking(picking, with_progress=True)
+        if with_lines:
+            data.update(
+                {"move_lines": self._data_for_move_lines(picking.move_line_ids)}
+            )
+        return data
+
+    def _data_for_stock_pickings(self, pickings, with_lines=False):
```

Comment:
When showing the lines of a reception, we should show the moves and not the move lines.